File: angular-ecomm/src/app/common/order-item.ts
import { CartItem } from "./cart-item";

export class OrderItem {
   

        // constructor(public imageUrl: string,
        //             public unitPrice: number,
        //             public quantity: number,
        //             public productId: string) { }
    
    
    
    imageUrl: string;
    unitPrice: number;
    quantity: number;
    productId: number;

    constructor(cartItem: CartItem)
    {
        this.imageUrl = cartItem.imageUrl;
        this.quantity= cartItem.quantity;
        this.unitPrice= cartItem.unitPrice;
        this.productId = cartItem.id;
    }

}

File: spring-boot-ecommerce/src/main/java/com/luv2code/ecommerce/entity/OrderItem.java
package com.luv2code.ecommerce.entity;

import lombok.Getter;
import lombok.Setter;

import javax.persistence.*;

@Entity
@Table(name="order_item")
@Getter
@Setter
public class OrderItem {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    @Column(name="id")
    private Long id;
    @Column(name="image_url")
    private String imageUrl;
    @Column(name="unit_price")
    private String unitPrice;
    @Column(name="quantity")
    private String quantity;
    @Column(name="product_id")
    private Long productId;
    @ManyToOne
    @JoinColumn(name = "order_id")
    private Order order;

}
